index.js

```diff
@@ -427,28 +448,41 @@ const clearOldFile = async (fileContract, fileName, hexName, chunkLength) => {
 
   if (oldChunkLength > chunkLength) {
     // remove
-    const option = {nonce: getNonce()};
-    let tx;
-    try {
-      tx = await fileContract.remove(hexName, option);
-    } catch (e) {
-      await sleep(3000);
-      tx = await fileContract.remove(hexName, option);
-    }
-    console.log(`Remove Transaction Id: ${tx.hash}`);
-    const receipt = await getTxReceipt(fileContract, tx.hash);
-    if (receipt.status) {
-      console.log(`Remove file: ${fileName}`);
-      return REMOVE_SUCCESS;
-    } else {
-      return REMOVE_FAIL;
-    }
+    return removeFile(fileContract, fileName, hexName);
   }
   return REMOVE_NORMAL;
 }
 // **** utils ****
+const checkBalance = async (provider, domainAddr, accountAddr) => {
+  return Promise.all([provider.getBalance(domainAddr), provider.getBalance(accountAddr)]).then(values => {
+    return {
+      domainBalance: values[0],
+      accountBalance: values[1]
+    };
+  }, reason => {
+    console.log(reason);
+  });
+}
 
 // **** function ****
+const remove = async (fileName, domain, key, RPC) => {
+  const {providerUrl, chainId, address} = await getWebHandler(domain, RPC);
+  if (providerUrl && parseInt(address) > 0) {
+    const provider = new ethers.providers.JsonRpcProvider(providerUrl);
+    const wallet = new ethers.Wallet(key, provider);
+    const fileContract = new ethers.Contract(address, fileAbi, wallet);
+    let prevInfo;
+    await checkBalance(provider, domain, wallet.address).then(info => { prevInfo = info; })
```

Comment:
'domain may be the ens domain name, this should use the 'address' returned by webHandler.